### PR TITLE
when running event.send (or any fire_master), dont destroy io_loop

### PR DESCRIPTION
**Pull request by mattp-**
### What does this PR do?
without this param, the SaltEvent object closes the shared io_loop running the
minion itself, causing a hang.

### Previous Behavior
in multimaster, the minion itself will hang if this occurred, causing minion unresponsiveness.

### New Behavior
minion handles connection loss / event.send behavior properly.

### Tests written?
no

### Commits signed with GPG?
no

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by mattp-** [comment]
this might be worthy of a backport to each release branch given the inertness vs the bug fix.

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
Nice fix here, @mattp-

**Comment by mattp-** [comment]
please don't merge yet - dealing with some funky behavior with MinionEvent I'm trying to sort out. thanks

**Comment by gtmanfred** [comment]
@mattp- i added [WIP] to the title, so it will mark the WIP job as in progress, once you are done if you remove that, we will check again.

Thanks!
Daniel

**Review by gtmanfred — APPROVED** [review]

**Comment by cachedout** [comment]
@DmitryKuzmenko Can you dive into this as well, please?

File: salt/modules/event.py [hunk 1]
```diff
@@ -90,14 +90,16 @@ def fire_master(data, tag, preload=None):
...
+                    ret = False
         return ret
     else:
         # Usually, we can send the event via the minion, which is faster
         # because it is already authenticated
         try:
-            return salt.utils.event.MinionEvent(__opts__, listen=False).fire_event(
-                {'data': data, 'tag': tag, 'events': None, 'pretag': None}, 'fire_master')
+            me = salt.utils.event.MinionEvent(__opts__, listen=False, keep_loop=True)
+            me.fire_event({'data': data, 'tag': tag, 'events': None, 'pretag': None}, 'fire_master')
```
Comment (DmitryKuzmenko) [inline salt/modules/event.py]: @mattp- you've removed the `return` statement here. Just missed it? If not could you please comment this?

**Review by DmitryKuzmenko — APPROVED** [review]
I'm OK here with one minor question.

**Pull request merged**